### DISCUSSION TITLE
0.3.0 Dev Roadmap | 开发路线

### DISCUSSION THREAD

**Discussion by Antonoko** [Antonoko/Windrecorder#82] [Announcements 公告]
## Scope:
0.3.0 上线将会新增以下功能改动： 

- [ ] https://github.com/yuka-friends/Windrecorder/issues/26 优化数据库存取逻辑
- [ ] https://github.com/yuka-friends/Windrecorder/issues/54 重构录制模块，替换 ffmpeg 录屏；

---

预计解决 issue 清单与 bug 修正：

---

预计发版日期：不早于 202X.xx.xx

此版本路线项可能将会拆分调整到未来版本。

**Comment by ASC8384** [comment]
Which version is prepared to add related applications of the LLMs?

**Reply by Antonoko** [reply]
i am not sure, maybe create a simple playground firstly as extension? haven’t figured out how LLM should be accessed. If there is a local model that saves resources and supports a large context window, I think I can make a daily activity summary first. As for conversational question search, it seems that it is essentially just converting user intent into an existing search query. LLM cannot truly understand the data and can only summarize based on the recall results. It seems that there is not much improvement compared to manual search? In addition, the biggest problem may be how to deal with privacy data. It would be convenient if everything could run locally, or if the user had an edge computing server on the intranet.

---

没想好LLM应该以什么形式接入，如果有比较节省资源的、支持**很大上下文窗口**的本地小模型，感觉可以先做一个每日活动总结，以先作为demo 扩展的形态试试看用起来怎么样。至于对话提问搜寻，感觉本质上也只是把用户意图转换为现在已有的搜索query，llm不能真正理解数据，只能根据召回结果洗稿总结，好像相比手动搜索并没有太大提升？此外最大的问题可能还是如何处理隐私数据，如果一切能跑在本地，或者用户有一个内网的边缘计算服务器就方便了。（因为还没想好，我不大确定是否值得现阶段就做、或者有什么提效的场景可以讨论？（想着也许未来随着技术发展会更好实现，比如用chrome的Gemini nano就能完成计算、有很方便的小模型可以使用等等
